lib/geoblacklight/relation/relation_response.rb

```diff
@@ -1,19 +1,22 @@
 module Geoblacklight
   module Relation
     class RelationResponse
-      attr_reader :relations
+      attr_reader :id
       def initialize(id, repository)
         @search_id = id
         @repository = repository
-        @relations = {
-          ancestors: Geoblacklight::Relation::Ancestors.new(@search_id, @repository).results,
-          descendants: Geoblacklight::Relation::Descendants.new(@search_id, @repository).results,
-          current_doc: id
-        }
+      end
+
+      def ancestors
+        Geoblacklight::Relation::Ancestors.new(@search_id, @repository).results
+      end
+
+      def descendants
+        Geoblacklight::Relation::Descendants.new(@search_id, @repository).results
```

Comment:
We should memoize these reduce the calls to Solr. Right now, relation#index make 7 calls to solr. Two for ancestor and five for descendants.

` @descendants ||= Geoblacklight::Relation::Descendants.new(@search_id, @repository).results`